lib/vstd/DateUtils.cpp

```diff
@@ -1,33 +1,29 @@
 #include "StdInc.h"
 #include <vstd/DateUtils.h>
 
-#if defined(VCMI_ANDROID)
-#include "../CAndroidVMHelper.h"
-#endif
-
 VCMI_LIB_NAMESPACE_BEGIN
 
 namespace vstd
 {
 
-	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt)
+	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt, std::string format)
 	{
-#if defined(VCMI_ANDROID)
-		CAndroidVMHelper vmHelper;
-		return vmHelper.callStaticStringMethod(CAndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "getFormattedDateTime");
-#endif
-
 		std::tm tm = *std::localtime(&dt);
 		std::stringstream s;
-		try
-		{
-			s.imbue(std::locale(""));
-		}
-		catch(const std::runtime_error & e)
+		if(format.empty())
```

Comment:
Perhaps we should always explicitly specify format in caller code and remove this condition?

Reply:
considering that other_ languages will never get here, I agree with removing the condition